packages/core/src/app/Renderer.ts

```diff
@@ -298,12 +304,14 @@ export class Renderer {
     if (
       result === RendererResult.Success &&
       this.exporter &&
-      this.exporter.generateAudio
+      this.exporter.mergeMedia
     ) {
-      //only generate audio when rendering images was actually successful
-      const endFrame = Math.min(this.playback.duration, this.playback.frame);
+      //only merge media when rendering images was actually successful
       try {
-        await this.exporter.generateAudio(mediaAssets, endFrame);
+        if (generateAudioPromise) {
+          await generateAudioPromise;
+        }
+        await this.exporter.mergeMedia();
```

Comment:
As discussed, we could probably move mergeMedia into the if statement and then even turn the try/catch and if statement inside out.